dist/Data-Dumper/Dumper.xs

```diff
@@ -651,25 +653,34 @@ dump_regexp(pTHX_ SV *retval, SV *val)
 
     assert(sv_pattern);
 
-    if (SvUTF8(sv_pattern)) {
-        sv_utf8_upgrade(retval);
-    }
+    is_utf8 = SvUTF8(sv_pattern);
 
-    rval = SvPV(sv_pattern, rlen);
-    rend = rval+rlen;
-    slash = rval;
     sv_catpvs(retval, "qr/");
 
-    for ( ; slash < rend; slash++) {
-        if (*slash == '\\') {
-            ++slash;
-            continue;
+    rval = SvPV(sv_pattern, rlen);
+    rend = rval+rlen;
```

Comment:
To be clear, **this** patch doesn't change the ordering, but I thought that **in the general case** it was a subtle bad idea to use `SvUTF8` **before** `SvPV` (and friends) because on a tied variable, the latter will call the relevant tie magic, and **that** can change the state of `SVf_UTF8`.

(And IIRC i wrote deliberately evil tests somewhere once that had a tied variable that returned the same code point, but toggle which representation, to try to trigger this sort of problem.)

What I don't know (easily) is whether a tied value can get this far in, but I don't think that there's any harm here in re-ordering this code so that `SvUTF8` is later, and I'll do that in a followup commit.

Of course, I should then audit the rest of the code for this... it seems to use `DO_UTF8` and always does it after `SvPV`.